src/core/transaction.c

```diff
@@ -1102,6 +1102,12 @@ int transaction_add_job_and_dependencies(
         }
 
         return 0;
+fail:
+        /* Recursive call failed to add required jobs so let's drop top level job as well. */
+        log_unit_debug_errno(unit, r, "Cannot add dependency job: %s", bus_error_message(e, r));
+        transaction_delete_job(tr, ret, false);
```

Comment:
please prefix the `false` with `/* delete_dependencies= */` as we do these days for function parameters that otherwise are just too opaque to figure out what they might mean.